src/streams.lisp

```diff
@@ -112,7 +116,7 @@
           (ssl-stream-peeked-byte stream)
         (setf (ssl-stream-peeked-byte stream) nil))
       (handler-case
-          (let ((buf (ssl-stream-input-buffer stream))
+          (let ((buf (ssl-stream-input-buffer-small stream))
```

Comment:
Why reading into a single-byte buffer is better than reading one byte into the big buffer in stream-listen and stream-read-byte?

Reply:
Sorry I missed the commit message (I looked in the pull request conversation and in the code comments before asking though).

I see now in the commit message you adress that question.